components/camel-kafka/src/main/java/org/apache/camel/component/kafka/consumer/support/KafkaRecordProcessorFacade.java

```diff
@@ -126,7 +129,7 @@ private ProcessingResult processRecord(
 
         ProcessingResult currentResult
                 = kafkaRecordProcessor.processExchange(exchange, partition, partitionHasNext,
-                        recordHasNext, record, lastResult, camelKafkaConsumer.getExceptionHandler());
+                        recordHasNext, record, camelKafkaConsumer.getExceptionHandler());
 
         if (!currentResult.isBreakOnErrorHit()) {
             commitManager.recordOffset(partition, currentResult.getPartitionLastOffset());
```

Comment:
this part of the code is one of 2 reasons I can see that we need the offset in `ProcessedResult`
may be worth trying to move it inside `createExchange`